docs/bundle_format.rst

```diff
@@ -0,0 +1,106 @@
+Bundle Format
+================
+
+The Blockstore does not parse or validate most files itself. However, some conventions must be followed by clients
+to allow interoperability between environments:
+
+* There are no constraints on how the files in a Bundle can be organized.
+* OLX files must use the **.olx** extension. Files with a different extension may be ignored or not treated as
+  OLX by plugins and clients.
+* A bundle.json file must be added by authoring clients which specifies the shareable components, bundle
+  dependencies and assets. See below for details.
+* Every XBlock element in an OLX file must have an **id** attribute. The value of this attribute will be used
+  to construct the **block_key**. In a single Bundle, all ids must be unique. Blockstore will not validate ids however.
+* Nested XBlocks in separate OLX files can be referenced by adding a **path** attribute to the element which points
+  to the relevant OLX file. Blockstore does not parse or validate any paths itself. However plugins may do so.
+* XBlock Runtimes should provide a service which can be used to read a file given a **path** or to get a
+  public URL for it (if it is specified in assets field). XBlocks will be responsible for replacing paths
+  in any assets and resources they are using with URLs.
+
+Paths
+-------------------------
+
+Paths are URIs of the form **[//bundle_version_alias]path**.
```

Comment:
Add: "...where `path` starts with a `/`"  ?

Reply:
> Perhaps we can make all links relative instead of starting with //? If link convention is relative, then the links should still work if the Bundle is exported into a subdirectory somewhere on a user's machine.

Yeah that makes a lot of sense to me.

Reply:
I do see the proposal uses this convention. I thought about this quite a bit and landed on this variation with the dependency name as the domain name for the following reasons:

* We will have to reserve a root dir path for `links` and both at the Blockstore and client level add handling for "You cannot create a directory with this path" error messages.
* It applies to a subset of authoring situations since its not clear how the export-bundle-dependencies-into-subdirectory will work if the editing has to happen across multiple bundles. A more generic solution would be to have all bundles exported at the root of a single directory. And then, lets say, the XBlock workbench can load the correct files just like other LMS/Studio environments would. Maybe something like a directory backed Blockstore. Also note that even in course exports right now, there are no "correct paths".
* This solution also allows us to nicely support dependencies across Blockstores. For example, the author only needs to download the bundle they want to edit locally and for any dependencies not available locally the workbench can be configured to cache bundles from a remote Blockstore.

Would like your thoughts on the above!